qiskit_research/utils/pauli_twirling.py

```diff
@@ -109,19 +111,51 @@ def run(
     ) -> DAGCircuit:
         for run in dag.collect_runs(list(self.gates_to_twirl)):
             for node in run:
-                twirl_gates = TWIRL_GATES[node.op.name]
-                (before0, before1), (after0, after1) = twirl_gates[
-                    self.rng.integers(len(twirl_gates))
-                ]
-                mini_dag = DAGCircuit()
-                register = QuantumRegister(2)
-                mini_dag.add_qreg(register)
-                mini_dag.apply_operation_back(before0, [register[0]])
-                mini_dag.apply_operation_back(before1, [register[1]])
-                mini_dag.apply_operation_back(node.op, [register[0], register[1]])
-                mini_dag.apply_operation_back(after0, [register[0]])
-                mini_dag.apply_operation_back(after1, [register[1]])
-                dag.substitute_node_with_dag(node, mini_dag)
+                if len(node.op.params) == 1:  # supersede the parameterized list
```

Comment:
If we're supersedeing the parametrized list, then we should delete that list, or at least the entries that have been superseded. It seems like CNOT is being left alone, though I wonder if that could be included in this logic too...

Reply:
The CNOT is not parameterized, so the same logic does not belong here. Instead, I created a tuple of 2-qubit rotations gates for the `node.op` to check with `isinstance()`.